scripts/globals/weaponskills/energy_drain.lua

```diff
@@ -1,11 +1,35 @@
 -----------------------------------
 -- Energy Drain
 -----------------------------------
+require("scripts/globals/status")
+require("scripts/settings/main")
+require("scripts/globals/weaponskills")
+-----------------------------------
 local weaponskill_object = {}
 
 weaponskill_object.onUseWeaponSkill = function(player, target, wsID, tp, primary, action, taChar)
-   -- TODO: Should steal MP based on TP and not wake the mob involved.
-    return 1, 0, false, 65
+   -- TODO: Should not wake the mob involved. Not tested on sleeping mob
+
+    local damage = 60 -- Base mp absorbtion from http://wiki.ffo.jp/html/686.html saying it was between 60 and 120 and always higher than Energy Steal
+
+    -- params.ftp100 = 1 params.ftp200 = 1.5 params.ftp300 = 2 -- to follow earlier statement of been always higher than Energy Steal
+    -- params.mnd_wsc = 1
+
+    local damagemod = damage * ((tp - 1000) / 2000 + 1) -- Simplified formula for getting difference in tp return from 1000 tp to 3000 tp
+
+    if xi.settings.USE_ADOULIN_WEAPON_SKILL_CHANGES then
+        local attMind = player:getStat(xi.mod.MND)
+        damagemod = damagemod + attMind * 1
+    end
+
+    damagemod = damagemod * xi.settings.WEAPON_SKILL_POWER
+
+    if not target:isUndead() then
+        local mpReturn = target:addMP(-damagemod)
+        player:addMP(mpReturn)
+    end
```

Comment:
Couldnt this be simplified to:
```
    if not target:isUndead() then
        target:addMP(-damagemod)
        player:addMP(damagemod)
    end
```

Reply:
```int32 CBattleEntity::addMP(int32 mp)
{
    int32 cap = std::clamp(health.mp + mp, 0, GetMaxMP());
    mp        = health.mp - cap;
    health.mp = cap;
    if (mp != 0)
    {
        updatemask |= UPDATE_HP;
    }
    return abs(mp);
}```
It's done like that to prevent giving more MP than the mob has left

Reply:
now here I would have just done a check for mp with the undead check, or [clamped](https://github.com/LandSandBoat/server/blob/base/scripts/globals/utils.lua#L70) it depending if we needed mobs mp to just be nonzero or we needed the aspired amount to not be bigger than the remaining amount (its been years since I've had to care how energy drain worked lol) and aspir the spell is set up this way https://github.com/LandSandBoat/server/blob/base/scripts/globals/spells/black/aspir.lua#L45

¯\\\_(ツ)_/¯

Reply:
In this case the mp shouldn't be more than what the target as left.

Reply:
I could change it to this local player:addMP(target:addMP(-damagemod)) but I would need to test it since I don't know if this is actually proper coding and possible to do. I can't test right away but will later.

Reply:
After testing player:addMP(target:addMP(-damagemod)) still work as intended and lua as no problem with the syntax.
This is as simplified as I can make it.